test/project_types/script/layers/application/connect_app_test.rb

```diff
@@ -0,0 +1,130 @@
+          extension_point_type: extension_point_type,
+          script_name: script_name,
+          env: ShopifyCLI::Resources::EnvFile.new(
+            api_key: "api_key",
+            secret: "secret",
+            extra: { "UUID" => "uuid" }
+          )
+        )
+      end
+
+      it "should return false and not prompt" do
+        Script::Forms::RunAgainstShopifyOrg.expects(:ask).never
+        ShopifyCLI::Shopifolk.expects(:check).never
+        refute subject
+      end
+    end
+
+    describe "when script project env is not valid" do
+      def new_app(title, api_key, secret, app_type = "custom")
+        {
+          "title" => title,
+          "apiKey" => api_key,
+          "apiSecretKeys" => [{ "secret" => secret }],
+          "appType" => app_type,
+        }
+      end
+
+      def new_org(id, name)
+        {
+          "id" => id,
+          "businessName" => name,
+        }
+      end
+
+      let(:orgs) do
+        [
+          new_org(1, "business1"),
+          new_org(2, "business2"),
+        ]
+      end
+
+      let(:apps) do
+        [
+          new_app("app1", selected_api_key, selected_secret),
+          new_app("app2", selected_api_key, selected_secret),
+        ]
+      end
+
+      let(:existing_scripts) { [] }
+      let(:orgs_with_apps) do
+        orgs.map do |org|
+          org.merge({ "apps" => apps })
+        end
+      end
+      let(:selected_org) { orgs_with_apps.first }
+      let(:selected_app) { selected_org["apps"].first }
+      let(:selected_api_key) { "default_api_key" }
+      let(:selected_secret) { "default_secret" }
+      let(:selected_uuid) { nil }
+
+      let(:script_project) do
+        script_project_repository.create(
+          language: language,
+          extension_point_type: extension_point_type,
+          script_name: script_name,
+          env: ShopifyCLI::Resources::EnvFile.new(
+            api_key: selected_api_key,
+            secret: selected_secret,
+            # intentionally missing UUID
+          )
+        )
+      end
+
+      before do
+        ShopifyCLI::Shopifolk.stubs(check: false)
+        Script::Forms::RunAgainstShopifyOrg.stubs(:ask).returns(stub(response: false))
+
+        ShopifyCLI::PartnersAPI::Organizations
+          .stubs(:fetch_with_app)
+          .returns(orgs_with_apps)
+        Script::Forms::AskOrg.stubs(:ask).returns(stub(org: selected_org))
+
+        Script::Forms::AskApp.stubs(:ask).returns(stub(app: selected_app))
+
+        Script::Layers::Infrastructure::ScriptService
+          .any_instance
+          .stubs(:get_app_scripts)
+          .with(extension_point_type: extension_point_type)
+          .returns(existing_scripts)
+      end
+
+      it "should connect by calling create_env on script_project_repo" do
+        script_project_repository
+          .expects(:create_env)
+          .with(
+            api_key: selected_api_key,
+            secret: selected_secret,
+            uuid: selected_uuid
+          )
```

Comment:
Nitpick: why is this not in the `before` block?

Reply:
Hmmm I guess just to separate out the `stubs` vs `expects`.
The `before` is the setup of all the related things.
The `it ... do`, contains the end thing we are actually testing / expecting.